src/verilog/verilog_synthesis.cpp

```diff
@@ -113,6 +114,20 @@ exprt verilog_synthesist::from_bitvector(
 
     return struct_exprt{std::move(field_values), struct_type};
   }
+  else if(dest.id() == ID_union)
+  {
+    // We use the first field of the union.
```

Comment:
Just curious: does Verilog require that all fields have the same width? If not, could there be any problems if the first field wasn't sufficiently wide?

Reply:
Yes, in Verilog packed unions all fields must be the same width. Will add comment to say so.